dbusmock/templates/systemd-system.py

```diff
@@ -0,0 +1,29 @@
+'''systemd mock template
+
+This variant is for using on the system bus.
+
+It implements StartUnit, StartTransientUnit and StopUnit, but without starting
+or stopping anything.
+'''
+
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation; either version 3 of the License, or (at your option) any
+# later version.  See http://www.gnu.org/copyleft/lgpl.html for the full text
+# of the license.
+
+__author__ = 'Jonas Ådahl'
+__copyright__ = '(c) 2021 Red Hat'
+
+from dbusmock import mockobject
+
+systemd_base = mockobject.load_module('systemd-base')
+
+BUS_NAME = systemd_base.BUS_NAME
+MAIN_OBJ = systemd_base.MAIN_OBJ
+MAIN_IFACE = systemd_base.MAIN_IFACE
+SYSTEM_BUS = True
```

Comment:
That's the only place where the two differ, right? I suppose it would be nice if for these cases `spawn_server_template()` could be parameterized to launch the template on the system or session bus, e. g. by looking at `parameters.get('SYSTEM_BUS', module.SYSTEM_BUS)`, or by adding a new `system_bus=None` kwarg. Besides systemd, I can also think of libvirtd as a service which can legitimately appear on both buses.

If you think that will help you, I can get this in relatively quickly -- or of course feel free to include it in your branch.

Reply:
The only difference indeed is system vs session bus. If this could be part of the template instantiation, I could merge the three files into a single one, which would mean the `AddServiceMethod()` can be avoided while still being able to use decorations on the method.